web/book/tests/snapshots/snapshot__examples__employees-2.snap

```diff
@@ -14,34 +14,34 @@ WITH table_5 AS (
     e.emp_no,
     e.gender
 ),
-table_3 AS (
+table_5 AS (
   SELECT
-    AVG(table_4._expr_0) AS salary_avg,
-    STDDEV(table_4._expr_0) AS salary_sd,
+    AVG(table_2._expr_0) AS salary_avg,
+    STDDEV(table_2._expr_0) AS salary_sd,
     dm.emp_no
   FROM
-    table_5 AS table_4
-    JOIN dept_emp AS de ON table_4.emp_no = de.emp_no
+    table_4 AS table_2
```

Comment:
Irrelevant to this PR — one day we can collapse these!

Reply:
Actually a bit relevant.

Without this split, you'd have to assign a name to a table instance before knowing if it will compile to a subquery or a CTE.

Reply:
... which is not possible.

And now there is a time in compilation process where all table instances know whether they will be sub-queries or references to CTEs. So we can infer the instance name from the table reference.